src/apps/Permissions/Permissions.js

```diff
@@ -65,244 +35,177 @@ class Permissions extends React.Component {
       return {
         screen,
         address: data,
-        app: this.getAppByProxyAddress(data),
+        app: getAppByProxyAddress(data, apps),
         secondaryScreen,
         secondaryData,
       }
     }
-
-    if (screen === 'entity' && isAddress(data)) {
-      return { screen, address: data }
-    }
-
-    return home
   }
 
-  getAppByProxyAddress(proxyAddress) {
-    if (!proxyAddress) {
-      return null
-    }
-    return this.props.apps.find(app =>
-      addressesEqual(app.proxyAddress, proxyAddress)
-    )
-  }
-
-  goToHome = () => {
-    this.props.onParamsRequest(null)
-  }
-
-  handleOpenApp = proxyAddress => {
-    this.props.onParamsRequest(`app.${proxyAddress}`)
-  }
-
-  handleOpenEntity = address => {
-    if (this.getAppByProxyAddress(address)) {
-      return this.handleOpenApp(address)
-    }
-    this.props.onParamsRequest(`entity.${address}`)
-  }
-
-  handleManageRole = (proxyAddress, roleBytes) => {
-    this.props.onParamsRequest(`app.${proxyAddress}.role.${roleBytes}`)
-  }
-
-  createPermission = () => {
-    this.setState({ showAssignPermissionPanel: true })
-  }
-
-  closeAssignPermissionPanel = () => {
-    this.setState({ showAssignPermissionPanel: false })
-  }
+  return home
+}
 
-  closeManageRolePanel = () => {
-    const { params, onParamsRequest } = this.props
-    const location = this.getLocation(params)
+function Permissions({
+  apps,
+  appsLoading,
+  onParamsRequest,
+  params,
+  permissionsLoading,
+  wrapper,
+}) {
+  const { layoutName } = useLayout()
+  const [showAssignPermissionPanel, setShowAssignPermissionPanel] = useState(
+    false
+  )
+  const { resolveRole } = usePermissions()
+  const scrollTopElement = useRef(null)
+
+  // `params` should change every time we navigate into and out of a detailed
+  // permissions view, so this ensures the user starts at the top of the screen
+  // on every navigation change
+  useEffect(() => {
+    scrollTopElement.current.scrollIntoView()
+  }, [params])
+
+  const openHome = useCallback(() => {
+    onParamsRequest(null)
+  }, [onParamsRequest])
+
+  const openApp = useCallback(
+    proxyAddress => {
+      onParamsRequest(`app.${proxyAddress}`)
+    },
+    [onParamsRequest]
+  )
+
+  const manageRole = useCallback(
+    (proxyAddress, roleBytes) => {
+      onParamsRequest(`app.${proxyAddress}.role.${roleBytes}`)
```

Comment:
@bpierre @dizzypaty This has the perhaps surprising behaviour of bringing you into an app's detailed permissions view if you click on the context menu from the permissions home screen:

https://imgur.com/a/3a3Hjxg

From the code it looks like this was as intended, but I wanted to double check the behaviour.

Reply:
Would it make sense to update the role directly without going into the app's permissions screen?

Reply:
Yes, it might be a good idea to stay on the same screen. :+1:

Reply:
Done (with the move to normal paths) https://github.com/aragon/aragon/pull/899/commits/60b9314803d167fe584656af680ea02a0d6049b8